### ISSUE TITLE
Bug when building amplify backend

### ISSUE THREAD

**Issue by edwint88** [aws-amplify/amplify-cli#9209]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

curl

### If applicable, what version of Node.js are you using?

14.16

### Amplify CLI Version

7.6.2

### What operating system are you using?

mac

### Amplify Categories

auth, function, api

### Amplify Commands

Not applicable

### Describe the bug

After changing in prod env something and wanting to merge that to dev, it got me with this bug.
I even tried to override as described here: https://github.com/aws-amplify/amplify-cli/issues/9146

```
node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/resources/overrides-resource/tsconfig.json'
2021-12-03T18:45:38.832Z [WARNING]: ✔ Successfully pulled backend environment dev from the cloud.
2021-12-03T18:45:38.881Z [INFO]: 🛑 Build error : ENOENT: no such file or directory, open '/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/resources/overrides-resource/tsconfig.json'
2021-12-03T18:45:38.885Z [WARNING]: 2021-12-03T18:45:38.884Z - error: uncaughtException: Error: ENOENT: no such file or directory, open '/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/resources/overrides-resource/tsconfig.json' date=Fri Dec 03 2021 18:45:38 GMT+0000 (Coordinated Universal Time), pid=2170, uid=0, gid=0, cwd=/codebuild/output/src952531131/src/ng-website, execPath=/root/.nvm/versions/node/v12.21.0/bin/node, version=v12.21.0, argv=[/root/.nvm/versions/node/v12.21.0/bin/node, /root/.nvm/versions/node/v12.21.0/bin/amplify, pull, --amplify, {"envName":"dev","appId":"someappid"}, --providers, {"awscloudformation":{"configLevel":"project","useProfile":true,"profileName":"default"}}, --no-override, --no-codegen, --yes], rss=592723968, heapTotal=507797504, heapUsed=487657544, external=26795615, arrayBuffers=24742299, loadavg=[2.4, 1.09, 0.46], uptime=566, trace=[column=13, file=/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts, function=null, line=106, method=null, native=false, column=21, file=/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts, function=AmplifyAuthTransform.applyOverride, line=104, method=applyOverride, native=false, column=5, file=/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts, function=AmplifyAuthTransform.transform, line=72, method=transform, native=false, column=10, file=/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/utils/generate-auth-stack-template.ts, function=generateAuthStackTemplate, line=6, method=null, native=false], stack=[Error: Error: ENOENT: no such file or directory, open '/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/resources/overrides-resource/tsconfig.json',     at /root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts:106:13,     at AmplifyAuthTransform.applyOverride (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts:104:21),     at AmplifyAuthTransform.transform (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts:72:5),     at generateAuthStackTemplate (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/utils/generate-auth-stack-template.ts:6:10)]
2021-12-03T18:45:38.909Z [ERROR]: !!! Build failed
2021-12-03T18:45:38.909Z [ERROR]: !!! Non-Zero Exit Code detected
2021-12-03T18:45:38.910Z [INFO]: # Starting environment caching...
2021-12-03T18:45:38.910Z [INFO]: # Uploading environment cache artifact...
2021-12-03T18:45:38.997Z [INFO]: # Environment caching completed
```

### Expected behavior

to build the backend

### Reproduction steps

change something on prod (removed analytics for instance)
push amplify & git commit + push
change env to dev (both git & amplify)
git merge
amplify analytics remove
other changes
amplify push
git push

### Log output

`node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/resources/overrides-resource/tsconfig.json'
2021-12-03T18:45:38.832Z [WARNING]: ✔ Successfully pulled backend environment dev from the cloud.
2021-12-03T18:45:38.881Z [INFO]: 🛑 Build error : ENOENT: no such file or directory, open '/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/resources/overrides-resource/tsconfig.json'
2021-12-03T18:45:38.885Z [WARNING]: 2021-12-03T18:45:38.884Z - error: uncaughtException: Error: ENOENT: no such file or directory, open '/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/resources/overrides-resource/tsconfig.json' date=Fri Dec 03 2021 18:45:38 GMT+0000 (Coordinated Universal Time), pid=2170, uid=0, gid=0, cwd=/codebuild/output/src952531131/src/ng-website, execPath=/root/.nvm/versions/node/v12.21.0/bin/node, version=v12.21.0, argv=[/root/.nvm/versions/node/v12.21.0/bin/node, /root/.nvm/versions/node/v12.21.0/bin/amplify, pull, --amplify, {"envName":"dev","appId":"someappid"}, --providers, {"awscloudformation":{"configLevel":"project","useProfile":true,"profileName":"default"}}, --no-override, --no-codegen, --yes], rss=592723968, heapTotal=507797504, heapUsed=487657544, external=26795615, arrayBuffers=24742299, loadavg=[2.4, 1.09, 0.46], uptime=566, trace=[column=13, file=/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts, function=null, line=106, method=null, native=false, column=21, file=/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts, function=AmplifyAuthTransform.applyOverride, line=104, method=applyOverride, native=false, column=5, file=/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts, function=AmplifyAuthTransform.transform, line=72, method=transform, native=false, column=10, file=/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/utils/generate-auth-stack-template.ts, function=generateAuthStackTemplate, line=6, method=null, native=false], stack=[Error: Error: ENOENT: no such file or directory, open '/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/resources/overrides-resource/tsconfig.json',     at /root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts:106:13,     at AmplifyAuthTransform.applyOverride (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts:104:21),     at AmplifyAuthTransform.transform (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts:72:5),     at generateAuthStackTemplate (/root/.nvm/versions/node/v12.21.0/lib/node_modules/@aws-amplify/cli/node_modules/@aws-amplify/amplify-category-auth/src/provider-utils/awscloudformation/utils/generate-auth-stack-template.ts:6:10)]
2021-12-03T18:45:38.909Z [ERROR]: !!! Build failed
2021-12-03T18:45:38.909Z [ERROR]: !!! Non-Zero Exit Code detected
2021-12-03T18:45:38.910Z [INFO]: # Starting environment caching...
2021-12-03T18:45:38.910Z [INFO]: # Uploading environment cache artifact...
2021-12-03T18:45:38.997Z [INFO]: # Environment caching completed
`

### Additional information

I observed that also in the AWS console in the backend the Authentication tab is always loading.

**Comment by edwint88** [comment]
Trying to migrate: https://docs.amplify.aws/cli/migration/override/ I get following error:
`🛑 Build error : File '/**/node_modules/amplify-cli-core/resources/overrides-resource/tsconfig.json' was not included into executable at compilation stage. Please recompile adding it as asset or script.
Error: Error: File '/**/node_modules/amplify-cli-core/resources/overrides-resource/tsconfig.json' was not included into executable at compilation stage. Please recompile adding it as asset or script.
    at /snapshot/node_modules/@aws-amplify/amplify-category-auth/lib/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.js:48:23
    at AmplifyAuthTransform.applyOverride (/snapshot/node_modules/@aws-amplify/amplify-category-auth/lib/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.js:46:29)
    at AmplifyAuthTransform.transform (/snapshot/node_modules/@aws-amplify/amplify-category-auth/lib/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.js:333:9)
    at generateAuthStackTemplate (/snapshot/node_modules/@aws-amplify/amplify-category-auth/lib/provider-utils/awscloudformation/utils/generate-auth-stack-template.js:7:12)
    at /snapshot/node_modules/@aws-amplify/amplify-category-auth/lib/provider-utils/awscloudformation/handlers/resource-handlers.js:116:13
    at Object.run (/snapshot/node_modules/@aws-amplify/amplify-category-auth/lib/commands/auth/update.js:91:40)
    at Object.executeAmplifyCommand (/snapshot/node_modules/@aws-amplify/amplify-category-auth/lib/index.js:336:5)
    at executePluginModuleCommand (/snapshot/node_modules/@aws-amplify/cli/lib/execution-manager.js:187:5)
    at executeCommand (/snapshot/node_modules/@aws-amplify/cli/lib/execution-manager.js:35:9)
    at Object.run (/snapshot/node_modules/@aws-amplify/cli/lib/index.js:162:9)
🛑 There was an error updating the auth resource`

So it's a problem with the migration (that maybe last time I've ignored it)

**Comment by edwint88** [comment]
when I try to add a new env:
`🛑 Build error : File '/**/node_modules/amplify-cli-core/resources/overrides-resource/tsconfig.json' was not included into executable at compilation stage. Please recompile adding it as asset or script.
Error: Error: Error: File '/**/node_modules/amplify-cli-core/resources/overrides-resource/tsconfig.json' was not included into executable at compilation stage. Please recompile adding it as asset or script.
    at transformRootStack (/snapshot/node_modules/amplify-provider-awscloudformation/lib/override-manager/index.js:35:11)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)`

**Comment by parvusville** [comment]
I'm encountering the same problem, getting the same error message apparently with anything else than amplify status. Upgrading from 7.5.4 to 7.6.2 didn't fix it.

**Comment by edwint88** [comment]
I've spent 3 days hopping something is happening and in the end I've went back to 7.5.4 and done as said here: https://github.com/aws-amplify/amplify-cli/issues/9146 override all categories and finally worked

**Comment by parvusville** [comment]
Migrating to new transformer and adding the override fixed the issue for me too, using 7.6.2
`amplify api migrate
amplify override api`
